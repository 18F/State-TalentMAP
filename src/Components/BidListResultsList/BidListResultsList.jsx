import React from 'react';
import PropTypes from 'prop-types';
import BidListResultsCard from '../BidListResultsCard';
// import * as PROP_TYPES from '../../Constants/PropTypes';

const BidListResultsList = ({ bidList, toggleBidPosition }) => (
  <div className="usa-grid-full">
    {
      bidList.results.slice().map(bid => (
        <BidListResultsCard
          bid={bid}
          toggleBidPosition={toggleBidPosition}
          key={bid.id}
        />
      ))
    }
  </div>
  );

BidListResultsList.propTypes = {
  bidList: PropTypes.shape({}).isRequired,
  toggleBidPosition: PropTypes.func.isRequired,
};

export default BidListResultsList;
