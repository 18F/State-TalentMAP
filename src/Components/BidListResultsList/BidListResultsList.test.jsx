import { shallow } from 'enzyme';
import React from 'react';
import toJSON from 'enzyme-to-json';
import BidListResultsList from './BidListResultsList';

describe('BidListResultsListComponent', () => {
  const bidList = { results: [{ id: 1 }] };
  it('is defined', () => {
    const wrapper = shallow(
      <BidListResultsList
        bidList={bidList}
        toggleBidPosition={() => {}}
      />,
    );
    expect(wrapper).toBeDefined();
  });

  it('can receive props', () => {
    const wrapper = shallow(
      <BidListResultsList
        bidList={bidList}
        toggleBidPosition={() => {}}
      />,
    );
    expect(wrapper.instance().props.bidList)
      .toBe(bidList);
  });

  it('matches snapshot', () => {
    const wrapper = shallow(
      <BidListResultsList
        bidList={bidList}
        toggleBidPosition={() => {}}
      />,
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
