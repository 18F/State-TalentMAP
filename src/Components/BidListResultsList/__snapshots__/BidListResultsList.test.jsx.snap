// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BidListResultsListComponent matches snapshot 1`] = `
<div
  className="usa-grid-full"
>
  <BidListResultsCard
    bid={
      Object {
        "id": 1,
      }
    }
    condensedView={false}
    submitBid={[Function]}
    toggleBidPosition={[Function]}
  />
</div>
`;
